components/lifter.py

```diff
@@ -21,47 +23,51 @@ class Lifter:
     BOTTOM_HEIGHT = 0
     THRESHOLD = 0.005
 
-    HEIGHT_FROM_FLOOR = 0.2585  # height from floor to initial lift pos when reset in m
+    HEIGHT_FROM_FLOOR = 0.17  # height from floor to initial lift pos when reset in m
     CONTAINMENT_SIZE = 0  # height needed for the mechanism to work properly in m
 
-    UPPER_SCALE = 1.8288 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE  # in m
-    BALANCED_SCALE = 1.524 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
-    LOWER_SCALE = 1.2192 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
-    SWITCH = 0.47625 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
+    CUBE_HEIGHT = 0.4
+
+    UPPER_SCALE = 1.8288 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE + CUBE_HEIGHT  # in m
+    BALANCED_SCALE = 1.524 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE + CUBE_HEIGHT
+    LOWER_SCALE = 1.2192 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE + CUBE_HEIGHT
+    # SWITCH = 0.47625 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
+    SWITCH = 1.0 - HEIGHT_FROM_FLOOR + CONTAINMENT_SIZE
+
+    UPWARD_ACCELERATION = FREE_SPEED // 2
+    DOWNWARD_ACCELERATION = FREE_SPEED // 2
 
     def setup(self):
         """This is called after variables are injected by magicbot."""
         self.set_pos = None
 
         self.motor.setQuadraturePosition(0, timeoutMs=10)
         self.motor.setInverted(True)
-        self.motor.setNeutralMode(ctre.WPI_TalonSRX.NeutralMode.Brake)
-
-        self.motor.configPeakCurrentLimit(50, timeoutMs=10)
-        self.motor.configContinuousCurrentLimit(40, timeoutMs=10)
-        self.motor.enableCurrentLimit(True)
+        self.motor.setNeutralMode(ctre.NeutralMode.Brake)
 
         self.motor.overrideLimitSwitchesEnable(False)
-        self.motor.configReverseLimitSwitchSource(ctre.WPI_TalonSRX.LimitSwitchSource.FeedbackConnector, ctre.WPI_TalonSRX.LimitSwitchNormal.NormallyOpen, deviceID=0, timeoutMs=10)
-        self.motor.configForwardLimitSwitchSource(ctre.WPI_TalonSRX.LimitSwitchSource.Deactivated, ctre.WPI_TalonSRX.LimitSwitchNormal.Disabled, deviceID=0, timeoutMs=10)
+        self.motor.configReverseLimitSwitchSource(ctre.talonsrx.TalonSRX.LimitSwitchSource.FeedbackConnector, ctre.talonsrx.TalonSRX.LimitSwitchNormal.NormallyOpen, deviceID=0, timeoutMs=10)
+        self.motor.configForwardLimitSwitchSource(ctre.talonsrx.TalonSRX.LimitSwitchSource.Deactivated, ctre.talonsrx.TalonSRX.LimitSwitchNormal.Disabled, deviceID=0, timeoutMs=10)
```

Comment:
Oh boy. Luckily these enums are now exported as `ctre.LimitSwitchSource` and `ctre.LimitSwitchNormal`. I guess you never notice how verbose these APIs are until you see this…